pkg/controller/kogitoapp/kogitoapp_controller.go

```diff
@@ -327,3 +332,67 @@ func getKubernetesResources(kogitoRes *kogitores.KogitoAppResources) []utilsres.
 
 	return k8sRes
 }
+
+// Ensure that all Kogito images are available before the build.
+func (r *ReconcileKogitoApp) ensureKogitoImageStream(instance *v1alpha1.KogitoApp) (reconcile.Result, error) {
+	isISPresentOnOpenshiftNamespace := true
+	isImagestreamReady := true
+
+	imageStreamTag := kogitores.ImageStreamTag
+	if instance.Spec.Build.ImageS2I.ImageStreamTag != "" {
+		imageStreamTag = instance.Spec.Build.ImageS2I.ImageStreamTag
+		log.Infof("Using ImageS2I Tag from CR: %s", imageStreamTag)
+	}
+
+	// First look on the current namespace, if the imagestreams are available, return
+	// and don't check for anything else.
+	// look for the Kogito image stream on the default namespace (openshift)
+	// All needed images must be present, if not create it on current namespace
+	for _, key := range kogitores.ImageStreamNameList {
+		is, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: instance.Namespace}, imageStreamTag)
+		isOcpNs, _ := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: key, Namespace: kogitores.ImageStreamNamespace}, imageStreamTag)
+		if is == nil {
+			isImagestreamReady = false
+		}
+		if isOcpNs == nil {
+			isISPresentOnOpenshiftNamespace = false
+		}
+	}
+
+	if !isISPresentOnOpenshiftNamespace && !isImagestreamReady {
+
+		for _, is := range kogitores.KogitoImageStream(instance.Namespace, imageStreamTag, instance.Spec.Runtime, instance.Spec.Build.Native).Items {
+			is.Namespace = instance.Namespace
+			hasIs, err := openshift.ImageStreamC(r.client).FetchTag(types.NamespacedName{Name: is.Name, Namespace: instance.Namespace}, imageStreamTag)
+			if err != nil {
+				log.Error(err.Error())
+			}
+
+			if hasIs == nil {
+				_, err := openshift.ImageStreamC(r.client).CreateImageStream(&is)
```

Comment:
When the second reconcile loop reaches here, there's an error reporting that the IS is already created? Or we could just fetch first and create after.